diff -Nrup netkit-telnet-0.17.orig/telnet/Makefile netkit-telnet-0.17/telnet/Makefile
--- netkit-telnet-0.17.orig/telnet/Makefile	1999-08-01 07:06:37.000000000 +0200
+++ netkit-telnet-0.17/telnet/Makefile	2014-06-15 09:32:50.136000000 +0200
@@ -1,13 +1,15 @@
 all: telnet
 
-include ../MCONFIG
-include ../MRULES
+#include ../MCONFIG
+#include ../MRULES
 
 #CXXFLAGS:=$(patsubst -O2, -g, $(CXXFLAGS))
 
+CXX ?= g++
 # -DAUTHENTICATE
-CXXFLAGS += -DUSE_TERMIO -DKLUDGELINEMODE
-LIBS += $(LIBTERMCAP)
+CXXFLAGS += -DUSE_TERMIO -DKLUDGELINEMODE -trigraphs
+LIBS += -lncurses
+LDFLAGS += -static
 
 SRCS = commands.cc main.cc network.cc ring.cc sys_bsd.cc telnet.cc \
 	terminal.cc tn3270.cc utilities.cc genget.cc environ.cc netlink.cc
diff -Nrup netkit-telnet-0.17.orig/telnet/commands.cc netkit-telnet-0.17/telnet/commands.cc
--- netkit-telnet-0.17.orig/telnet/commands.cc	2000-07-23 06:16:24.000000000 +0200
+++ netkit-telnet-0.17/telnet/commands.cc	2014-06-15 09:29:38.336000000 +0200
@@ -64,6 +64,7 @@ char cmd_rcsid[] =
 
 #include "ring.h"
 
+#define _POSIX_VDISABLE 1
 #include "externs.h"
 #include "defines.h"
 #include "types.h"
@@ -88,7 +89,7 @@ char cmd_rcsid[] =
 
 #ifndef       MAXHOSTNAMELEN
 #define       MAXHOSTNAMELEN 64
-#endif        MAXHOSTNAMELEN
+#endif        /* MAXHOSTNAMELEN */
 
 #if	defined(HAS_IPPROTO_IP) && defined(IP_TOS)
 int tos = -1;
@@ -1337,8 +1338,8 @@ int shell(int argc, const char **) {
 
 	    shellp = getenv("SHELL");
 	    if (shellp == NULL)
-		shellp = "/bin/sh";
-	    if ((shellname = rindex(shellp, '/')) == 0)
+		shellp = "/system/bin/sh";
+	    if ((shellname = strrchr(shellp, '/')) == 0)
 		shellname = shellp;
 	    else
 		shellname++;
diff -Nrup netkit-telnet-0.17.orig/telnet/environ.cc netkit-telnet-0.17/telnet/environ.cc
--- netkit-telnet-0.17.orig/telnet/environ.cc	2000-07-23 05:24:53.000000000 +0200
+++ netkit-telnet-0.17/telnet/environ.cc	2014-06-15 09:20:10.656000000 +0200
@@ -5,6 +5,7 @@
 #include <arpa/telnet.h>
 #include "ring.h"
 #include "defines.h"
+#define _POSIX_VDISABLE 1
 #include "externs.h"
 #include "environ.h"
 #include "array.h"
diff -Nrup netkit-telnet-0.17.orig/telnet/main.cc netkit-telnet-0.17/telnet/main.cc
--- netkit-telnet-0.17.orig/telnet/main.cc	1999-08-01 07:06:37.000000000 +0200
+++ netkit-telnet-0.17/telnet/main.cc	2014-06-15 09:20:10.656000000 +0200
@@ -46,8 +46,10 @@ char main_rcsid[] =
 #include <sys/types.h>
 #include <getopt.h>
 #include <string.h>
+#include <stdlib.h>
 
 #include "ring.h"
+#define _POSIX_VDISABLE 1
 #include "externs.h"
 #include "defines.h"
 #include "proto.h"
diff -Nrup netkit-telnet-0.17.orig/telnet/network.cc netkit-telnet-0.17/telnet/network.cc
--- netkit-telnet-0.17.orig/telnet/network.cc	1996-08-13 10:09:58.000000000 +0200
+++ netkit-telnet-0.17/telnet/network.cc	2014-06-15 09:20:10.666000000 +0200
@@ -42,9 +42,11 @@ char net_rcsid[] =
 #include <sys/time.h>
 #include <errno.h>
 #include <arpa/telnet.h>
+#include <stdlib.h>
 
 #include "ring.h"
 #include "defines.h"
+#define _POSIX_VDISABLE 1
 #include "externs.h"
 #include "proto.h"
 #include "netlink.h"
diff -Nrup netkit-telnet-0.17.orig/telnet/sys_bsd.cc netkit-telnet-0.17/telnet/sys_bsd.cc
--- netkit-telnet-0.17.orig/telnet/sys_bsd.cc	1999-09-28 18:29:24.000000000 +0200
+++ netkit-telnet-0.17/telnet/sys_bsd.cc	2014-06-15 09:20:10.666000000 +0200
@@ -56,6 +56,7 @@ char bsd_rcsid[] =
 #include "ring.h"
 
 #include "defines.h"
+#define _POSIX_VDISABLE 1
 #include "externs.h"
 #include "types.h"
 #include "proto.h"
diff -Nrup netkit-telnet-0.17.orig/telnet/telnet.cc netkit-telnet-0.17/telnet/telnet.cc
--- netkit-telnet-0.17.orig/telnet/telnet.cc	2000-07-23 05:24:53.000000000 +0200
+++ netkit-telnet-0.17/telnet/telnet.cc	2014-06-15 09:31:44.226000000 +0200
@@ -49,6 +49,7 @@ char telnet_rcsid[] =
 
 #include "ring.h"
 #include "defines.h"
+#define _POSIX_VDISABLE 1
 #include "externs.h"
 #include "types.h"
 #include "environ.h"
@@ -61,10 +62,10 @@ char telnet_rcsid[] =
  * Due to lossage in some linux distributions/kernel releases/libc versions
  * this must come *after* termios.h (which is included in externs.h)
  */
-#include <termcap.h>
+#include <ncurses/termcap.h>
 
 #ifdef USE_NCURSES
-#include <term.h>
+#include <ncurses/term.h>
 #endif
 
 
diff -Nrup netkit-telnet-0.17.orig/telnet/terminal.cc netkit-telnet-0.17/telnet/terminal.cc
--- netkit-telnet-0.17.orig/telnet/terminal.cc	1999-12-12 20:48:05.000000000 +0100
+++ netkit-telnet-0.17/telnet/terminal.cc	2014-06-15 09:30:25.086000000 +0200
@@ -45,9 +45,11 @@ char terminal_rcsid[] =
 #include <signal.h>
 #include <errno.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 #include "ring.h"
 #include "defines.h"
+#define _POSIX_VDISABLE 1
 #include "externs.h"
 #include "types.h"
 #include "proto.h"
@@ -468,7 +470,7 @@ void TerminalNewMode(int f)
 
 #ifdef	SIGINFO
 	signal(SIGINFO, ayt);
-#endif	SIGINFO
+#endif	/* SIGINFO */
 
 #if defined(NOKERNINFO)
 	tmp_tc.c_lflag |= NOKERNINFO;
@@ -504,7 +506,7 @@ void TerminalNewMode(int f)
 
 #ifdef	SIGINFO
 	signal(SIGINFO, ayt_status);
-#endif	SIGINFO
+#endif	/* SIGINFO */
 
 #ifdef	SIGTSTP
 	signal(SIGTSTP, SIG_DFL);
diff -Nrup netkit-telnet-0.17.orig/telnet/tn3270.cc netkit-telnet-0.17/telnet/tn3270.cc
--- netkit-telnet-0.17.orig/telnet/tn3270.cc	1996-08-13 11:08:34.000000000 +0200
+++ netkit-telnet-0.17/telnet/tn3270.cc	2014-06-15 09:20:10.696000000 +0200
@@ -42,6 +42,7 @@ char tn3270_rcsid[] =
 
 #include "defines.h"
 #include "ring.h"
+#define _POSIX_VDISABLE 1
 #include "externs.h"
 #include "proto.h"
 
diff -Nrup netkit-telnet-0.17.orig/telnet/utilities.cc netkit-telnet-0.17/telnet/utilities.cc
--- netkit-telnet-0.17.orig/telnet/utilities.cc	1999-12-12 16:33:40.000000000 +0100
+++ netkit-telnet-0.17/telnet/utilities.cc	2014-06-15 09:20:10.706000000 +0200
@@ -47,9 +47,11 @@ char util_rcsid[] =
 #include <sys/socket.h>
 #include <unistd.h>
 #include <ctype.h>
+#include <stdlib.h>
 
 #include "ring.h"
 #include "defines.h"
+#define _POSIX_VDISABLE 1
 #include "externs.h"
 #include "proto.h"
 #include "terminal.h"
