diff -Nrup olsrd-0.6.6.2.orig/Makefile olsrd-0.6.6.2/Makefile
--- olsrd-0.6.6.2.orig/Makefile	2014-06-02 12:15:27.000000000 +0200
+++ olsrd-0.6.6.2/Makefile	2014-06-10 18:09:10.147000116 +0200
@@ -203,7 +203,7 @@ ifeq ($(OS),win32)
 SUBDIRS := dot_draw httpinfo jsoninfo mini pgraph secure txtinfo
 else
 ifeq ($(OS),android)
-SUBDIRS := arprefresh bmf dot_draw dyn_gw dyn_gw_plain httpinfo jsoninfo mdns mini nameservice p2pd pgraph pud secure sgwdynspeed txtinfo watchdog
+SUBDIRS := arprefresh bmf dot_draw dyn_gw dyn_gw_plain httpinfo jsoninfo mdns mini nameservice p2pd pgraph pud quagga secure sgwdynspeed txtinfo watchdog
 else
 SUBDIRS := dot_draw httpinfo jsoninfo mini nameservice pgraph secure txtinfo watchdog
 endif
diff -Nrup olsrd-0.6.6.2.orig/lib/quagga/src/quagga.h olsrd-0.6.6.2/lib/quagga/src/quagga.h
--- olsrd-0.6.6.2.orig/lib/quagga/src/quagga.h	2014-06-02 12:15:27.000000000 +0200
+++ olsrd-0.6.6.2/lib/quagga/src/quagga.h	2014-06-10 18:10:50.266998327 +0200
@@ -21,7 +21,7 @@
 
 /* Zebra socket */
 #ifndef ZEBRA_SOCKPATH
-#define ZEBRA_SOCKPATH "/var/run/quagga/zserv.api"
+#define ZEBRA_SOCKPATH "/data/local/var/run/quagga/zserv.api"
 #endif /* ZEBRA_SOCKPATH */
 
 /* Quagga plugin flags */
