diff -Nrup quagga-0.99.21mr2.2.orig/lib/log.c quagga-0.99.21mr2.2/lib/log.c
--- quagga-0.99.21mr2.2.orig/lib/log.c	2013-07-24 22:04:59.103079265 +0200
+++ quagga-0.99.21mr2.2/lib/log.c	2013-07-24 22:05:11.203139266 +0200
@@ -941,7 +941,13 @@ proto_redistnum(int afi, const char *s)
 	return ZEBRA_ROUTE_ISIS;
       else if (strncmp (s, "bg", 2) == 0)
 	return ZEBRA_ROUTE_BGP;
-      else if (strncmp (s, "ba", 2) == 0)
+      else if (strncmp (s, "h", 1) == 0)
+	return ZEBRA_ROUTE_HSLS;
+      else if (strncmp (s, "ol", 2) == 0)
+	return ZEBRA_ROUTE_OLSR;
+      else if (strncmp (s, "bat", 3) == 0)
+	return ZEBRA_ROUTE_BATMAN;
+      else if (strncmp (s, "bab", 3) == 0)
 	return ZEBRA_ROUTE_BABEL;
     }
   if (afi == AFI_IP6)
