diff -Nrup quagga-0.99.22.4.orig/lib/command.c quagga-0.99.22.4/lib/command.c
--- quagga-0.99.22.4.orig/lib/command.c	2013-08-04 15:01:38.000000000 +0200
+++ quagga-0.99.22.4/lib/command.c	2014-06-14 15:16:31.353134687 +0200
@@ -522,7 +522,6 @@ zencrypt (const char *passwd)
 {
   char salt[6];
   struct timeval tv;
-  char *crypt (const char *, const char *);
 
   gettimeofday(&tv,0);
   
@@ -530,7 +529,7 @@ zencrypt (const char *passwd)
   to64(&salt[3], tv.tv_usec, 3);
   salt[5] = '\0';
 
-  return crypt (passwd, salt);
+  return passwd;
 }
 
 /* This function write configuration of this host. */
diff -Nrup quagga-0.99.22.4.orig/lib/vty.c quagga-0.99.22.4/lib/vty.c
--- quagga-0.99.22.4.orig/lib/vty.c	2013-08-04 15:01:38.000000000 +0200
+++ quagga-0.99.22.4/lib/vty.c	2014-06-14 15:16:31.357134688 +0200
@@ -334,7 +334,6 @@ vty_auth (struct vty *vty, char *buf)
   char *passwd = NULL;
   enum node_type next_node = 0;
   int fail;
-  char *crypt (const char *, const char *);
 
   switch (vty->node)
     {
@@ -358,12 +357,7 @@ vty_auth (struct vty *vty, char *buf)
     }
 
   if (passwd)
-    {
-      if (host.encrypt)
-	fail = strcmp (crypt(buf, passwd), passwd);
-      else
 	fail = strcmp (buf, passwd);
-    }
   else
     fail = 1;
 
